src/spikeinterface/core/numpyextractors.py

```diff
@@ -337,6 +339,41 @@ def from_times_labels(times_list, labels_list, sampling_frequency, unit_ids=None
 
         return sorting
 
+    @staticmethod
+    def from_times_and_labels(times_list, labels_list, sampling_frequency, unit_ids=None) -> "NumpySorting":
+        """
+        Construct NumpySorting extractor from:
+          * an array of spike times (in s)
+          * an array of spike labels
+        In case of multisegment, it is a list of array.
+
+        Parameters
+        ----------
+        times_list : list of array (or array)
+            An array of spike samples
+        labels_list : list of array (or array)
+            An array of spike labels corresponding to the given times
+        unit_ids : list or None, default: None
+            The explicit list of unit_ids that should be extracted from labels_list
+            If None, then it will be np.unique(labels_list)
+        """
+        if isinstance(times_list, np.ndarray):
+            assert isinstance(labels_list, np.ndarray)
+            times_list = [times_list]
+            labels_list = [labels_list]
+
+        sample_list = [np.round(t * sampling_frequency).astype("int64") for t in times_list]
+        return NumpySorting.from_samples_and_labels(sample_list, labels_list, sampling_frequency, unit_ids)
+
+    @staticmethod
+    def from_times_labels(times_list, labels_list, sampling_frequency, unit_ids=None) -> "NumpySorting":
+        warnings.warn(
+            "from_times_labels is deprecated, use from_sample_and_labels instead",
```

Comment:
Could we get a removal version so we don't forget to take this out in a couple version?
```suggestion
            "from_times_labels is deprecated and will be removed in 0.104.0, use from_sample_and_labels instead",
```